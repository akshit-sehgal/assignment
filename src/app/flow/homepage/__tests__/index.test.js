import React from 'react';
import { shallow } from 'enzyme';
import HomePage from '../';

describe('Testing <HomePage />', () => {
    let props;
    beforeEach(() => {
        props = {
            history: {
                push: jest.fn()
            }
        }
    })
    it('should render the default view', () => {
        const wrapper = shallow(<HomePage {...props} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('Connect(Header)').length).toEqual(1);
        expect(wrapper.find('Connect(CarouselHolder)').length).toEqual(1);
        expect(wrapper.find('Connect(ProductListing)').length).toEqual(1);
        expect(wrapper.find('Footer').length).toEqual(1);
    });

});