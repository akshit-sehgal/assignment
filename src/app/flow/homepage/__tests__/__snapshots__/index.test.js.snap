// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <HomePage /> should render the default view 1`] = `
<div
  className="homepage-container"
>
  <Connect(Header) />
  <Connect(CarouselHolder) />
  <Connect(ProductListing)
    navigate={[Function]}
  />
  <Footer />
</div>
`;
